aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -121,8 +119,8 @@ Tensor isclose(const Tensor& self, const Tensor& other, double rtol, double atol
 
   // Computes equality closeness
   Tensor close = self == other;
-  if (equal_nan && self.is_floating_point()) {
-      close.__ior__((self != self).__iand__(other != other));
+  if (equal_nan && (self.is_floating_point() || self.is_complex())) {
+      close.__ior__(self.isnan().__iand__(other.isnan()));
```

Comment:
Although it is functionally equal, `self.isnan()` is a lot more legible than `self != self`

Reply:
I guess so. The compiler can probably optimize away the extra function call.